internal/provider/transfer_endpoint_object_storage.go

```diff
@@ -411,8 +412,9 @@ func (m *endpointObjectStorageSourceSettings) convert() (*transfer.EndpointSetti
 func (m *endpointObjectStorageSourceSettings) parse(e *endpoint.ObjectStorageSource) diag.Diagnostics {
 	var diags diag.Diagnostics
 
-	m.PathPattern = types.StringValue(e.GetPathPattern())
-
+	if v := e.GetPathPattern(); v != "" {
+		m.PathPattern = types.StringValue(v)
+	}
```

Comment:
Do you have plans for **unsetting** this field? The field is _Optional_.
Like user wants to remove it and here we need to set `m.PathPattern = nil`.
We can to do this in `else-clause` if needed.

Reply:
yeah, strictly speaking being able to unset it might make sense, will change it :)